sample/sample.go

```diff
@@ -64,6 +145,24 @@ func Importance(samples, weights []float64, target dist.LogProber, proposal dist
 	}
 }
 
+// Rejectioner is a wrapper around the Rejection sampling generation procedure.
+// If the rejection sampling fails during the call to Sample, all samples will
+// be set to math.NaN().
+type Rejectioner struct {
```

Comment:
Rejectioner could hold an `Err error`field that can be queried after the call to `Sample`.